### PR TITLE
layers: Consolidate PipelineStage and AccessMask bit operations.

### PR DESCRIPTION
**Pull request by jeremyg-lunarg**
Core and sync validation need to do similar operations on these
bitmasks, but were using separated code. Make sync_utils.{h,cpp}
for the common stuff and rely on generated code as much as
possible.

This change also moves all the generated lookup tables into a
.cpp file so that they don't get built twice.

Note that there's several other places in core_validation.cpp
that can be cleaned up further, such as ValidateStageMaskGsTsEnables().
This will happen in a later patch.

Change-Id: I615b2a37b4f2cbf2f655d822601708f161cc92f9

### CODE REVIEW

File: layers/generated/synchronization_validation_types.h [hunk 1]
```diff
@@ -38,10 +38,10 @@
...
 
-// clang-format off
-
 using SyncStageAccessFlags = std::bitset<64>;
 
+// clang-format off
+
 // Unique number for each  stage/access combination
 enum SyncStageAccessIndex {
     SYNC_DRAW_INDIRECT_INDIRECT_COMMAND_READ = 0,
```
Comment (jzulauf-lunarg) [inline layers/generated/synchronization_validation_types.h:47]: Do we want/need a "none" value for default initializers?

**Review by jzulauf-lunarg — APPROVED** [review]
LGTM when CI is happy -- I wouldn't mind adding a "NONE" SyncAccessIndex in this change... there are a couple of places we don't do default initialization because there isn't a none.

**Review by sfricke-samsung — COMMENTED** [review]
I haven't taken too much of a look into sync val in the past, but in terms of the goal of this, I can't find any issues

File: layers/generated/synchronization_validation_types.h (hunk 1 above)
Comment (jeremyg-lunarg) [reply layers/generated/synchronization_validation_types.h:47]: This is tricksy. Lets do it in a separate patch.

**Pull request merged**